<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>780</width>
    <height>582</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="OpenGLWidget" name="openGLWidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>781</width>
      <height>441</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>450</y>
      <width>761</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>0 - vShader1</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>1 - Flat</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>2 - Goraud</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>3 - Phong</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>4 - Normal</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="function1">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>480</y>
      <width>241</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string> (x^2*z^2)/5 - 2</string>
    </property>
   </widget>
   <widget class="QPushButton" name="function2">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>480</y>
      <width>241</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>sin(x) * z</string>
    </property>
   </widget>
   <widget class="QPushButton" name="function3">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>480</y>
      <width>241</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>1.25*sin(x+z)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="function4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>510</y>
      <width>241</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string> sin(x^2 + z^2)/2</string>
    </property>
   </widget>
   <widget class="QPushButton" name="function6">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>510</y>
      <width>241</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>sin(x)*cos(z)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="function5">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>510</y>
      <width>241</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>abs(x) + abs(z) - 2</string>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen_File">
   <property name="text">
    <string>Open File</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>openglwidget.h</header>
   <slots>
    <signal>statusBarMessage(QString)</signal>
    <signal>enableComboShaders(bool)</signal>
    <slot>showFileOpenDialog()</slot>
    <slot>changeShader(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>openGLWidget</sender>
   <signal>statusBarMessage(QString)</signal>
   <receiver>statusBar</receiver>
   <slot>showMessage(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>289</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>changeShader(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>enableComboShaders(bool)</signal>
   <receiver>comboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
